### PR TITLE
feat(PX-1850): Composing and Sending test coverage report to Pulse

### PR DESCRIPTION
**Pull request by NarendraSG**
Added script to send the test coverage report generated to Pulse.

A secret named __PULSE_HAMC_SECRET__  and a variable  __PULSE_URL__ needs to be created on the organization level.

## Type of change

Please check the relevant option.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration

- [ ] Tested the changed workflows in a repo

## Checklist:

- [ ] I have reviewed my code while raising this PR
- [ ] I have added comments to my code, particularly in hard-to-understand areas
- [ ] I have created a new README file named as `README-<name-of-the-workflow-file>.md` in case of new workflow addition, or made corresponding changes to the documentation with respect to the changes.
- [ ] My changes generate no new warnings and errors

### CODE REVIEW

File: README-nodejs-ci.md [hunk 1]
```diff
@@ -47,6 +49,7 @@ CI scans workflow for NodeJS code.
...
 | `ALLOWED_LICENSES` <a name="action_variable_ALLOWED_LICENSES"></a> | List of allowed package licenses. Enter each license in new line                                                                                                                            | Apache-2.0<br>MIT<br>BSD-2-Clause<br>No-Licence<br>Unlicense<br>LGPL-2.1
 | `COVERAGE_THRESHOLD` <a name="action_variable_COVERAGE_THRESHOLD"></a> | Specifies the minimum acceptable percentage for test coverage. If the test coverage falls below this threshold, the scan will fail.                                                                                                                     |     `70` | no       |
+| `PULSE_URL`                                                            | Base URL of Pulse that will be used to send various generated reports to Pulse      | `https://a0lrn1xwl4.execute-api.eu-west-1.amazonaws.com`    | yes                                                                  |
```
[the remark this answers is not in the document]
Comment (NarendraSG) [reply README-nodejs-ci.md:52]: @mohankumar-studiographene This URL won't be changing as it is Pulse's backend URL where the data is sent thorugh POST request

File: .github/workflows/nodejs-ci.yml [hunk 2]
```diff
@@ -85,6 +91,16 @@ on:
         description: "Command to execute at the end of the container scan"
         type: string
         default: ""
+      send_test_coverage_report_to_pulse:
```
Comment (mohankumar-studiographene) [inline .github/workflows/nodejs-ci.yml:94]: Can we rename it to `send_dev_test_coverage_report_to_pulse` to keep in sync with other dev_test vars?

File: .github/workflows/nodejs-ci.yml [hunk 3]
```diff
@@ -85,6 +91,16 @@ on:
...
+        description: "A boolean value to enable sending test coverage report to Pulse"
+        type: boolean
+        required: false
+        default: false
+      test_coverage_report_directory:
```
Comment (mohankumar-studiographene) [inline .github/workflows/nodejs-ci.yml:99]: Can we rename it to `dev_test_coverage_report_directory` to keep in sync with other dev_test vars?

**Review by mohankumar-studiographene — CHANGES_REQUESTED** [review]

File: .github/workflows/nodejs-ci.yml [hunk 4]
```diff
@@ -6,7 +6,13 @@ on:
       NPM_TOKEN:
         description: "A GitHub PersonalAccessToken used to access GitHub package regististy. NPM_TOKEN is also passed as Docker build --build-arg. If not defined default GITHUB_TOKEN is used."
         required: false
+      PULSE_HAMC_SECRET:
+        description: "A secret key to generate HMAC for Pulse API. This is used to send test coverage report to Pulse."
+        required: false
     inputs:
+      PULSE_URL:
+        type: string
+        default: "https://a0lrn1xwl4.execute-api.eu-west-1.amazonaws.com"
```
[the remark this answers is not in the document]
Comment (NarendraSG) [reply .github/workflows/nodejs-ci.yml:15]: This is PULSE backend live server base url

**Review by aakashdeshpandesg — APPROVED** [review]

**Review by aakashdeshpandesg — APPROVED** [review]

File: .github/workflows/nodejs-ci.yml (hunk 2 above)
Comment (NarendraSG) [reply .github/workflows/nodejs-ci.yml]: YEs, did the changes

File: .github/workflows/nodejs-ci.yml (hunk 3 above)
Comment (NarendraSG) [reply .github/workflows/nodejs-ci.yml]: yes done

**Pull request merged**